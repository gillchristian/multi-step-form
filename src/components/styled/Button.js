import styled from 'styled-components'

export default styled.button`
  display: flex;
  align-items: center;
  justify-content: center;

  font-size: 1rem;
  margin-top: 1rem;
  margin-bottom: 1rem;
  padding: 0.25rem 1rem;
  color: inherit;

  appearance: none;

  background-color: transparent;
  border-width: 2px;
  border-style: solid;
  border-color: ${p => p.error ? 'palevioletred' : 'gray'};

  &:focus {
    outline: none;
  }

  &:hover{
    background-color: ${p => p.loading ? 'transparent' : 'gray'};
    cursor: ${p => p.loading ? 'wait' : 'pointer'};
  }
`
