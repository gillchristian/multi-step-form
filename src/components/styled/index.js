export { default as Center } from './Center'

export { default as Checkbox } from './Checkbox'
export { default as Toggle } from './Toggle'
export { default as Input } from './Input'
export { default as Button } from './Button'

export { default as Text } from './Text'
export { default as Help } from './Help'
export { default as Margin } from './Margin'
export { default as Spinner } from './Spinner'
export { default as Progress } from './Progress'
