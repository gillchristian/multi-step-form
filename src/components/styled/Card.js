import styled from 'styled-components'

export default styled.div`
  display: flex;
  flex-direction: column;
  margin: 0;
  padding: 10px;
  width: 80%;
  max-width: 500px;

  background-color: #f5f5f5;
  border-radius: 2px;
  box-shadow: 0 1px 1px rgba(0,0,0,0.15),
    0 10px 0 -5px #eee,
    0 10px 1px -4px rgba(0,0,0,0.15),
    0 20px 0 -10px #eee,
    0 20px 1px -9px rgba(0,0,0,0.15);

  overflow: hidden;
`
