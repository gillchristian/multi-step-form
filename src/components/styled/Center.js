import styled from 'styled-components'

export default styled.div`
  display: flex;
  justify-content: center;
  align-items: center;
  flex-direction: column;

  width: ${props => props.width ? props.width : 'auto'};
  height: ${props => props.height ? props.height : 'auto'};
`
