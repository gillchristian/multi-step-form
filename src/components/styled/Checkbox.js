import styled from 'styled-components'

// @link: https://goo.gl/IbWZxR
export default styled.div`
  position: relative;
  cursor: pointer;
  padding: 0;
  width: 1.2rem;
  height: 1.2rem;

  outline: none;
  user-select: none;

  &:before {
    content: '';
    margin-right: 10px;
    display: inline-block;
    vertical-align: text-top;
    width: 1.2rem;
    height: 1.2rem;
    background-color: ${p => p.checked ? '#f35429' : 'white' };
  }

  &:hover:before {
    background-color: #f35429;
  }

  &:after {
    display: ${p => p.checked ? 'block' : 'none'}
    content: '';
    position: absolute;
    left: 9px;
    top: 13px;
    background-color: white;
    width: 3px;
    height: 3px;
    box-shadow:
      2px 0 0 white,
      4px 0 0 white,
      4px -2px 0 white,
      4px -4px 0 white,
      4px -6px 0 white,
      4px -8px 0 white;
    transform: rotate(45deg);
  }
`
