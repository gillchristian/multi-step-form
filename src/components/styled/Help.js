import styled from 'styled-components'

const types = {
  success: '#3c763d',
  info: '#31708f',
  warnig: '#8a6d3b',
  danger: '#a94442',
}

export default styled.div`
  color: ${ p => types[p.type || 'info'] }
  font-style: italic;
  font-size: 0.8rem;
  margin: 5px;
`
