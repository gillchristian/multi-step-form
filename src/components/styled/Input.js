import styled from 'styled-components'

export default styled.input`
  border: none;
  border-bottom-width: 2px;
  border-bottom-style: solid;
  border-bottom-color: ${p => p.error ? 'palevioletred' : 'gray'};
  font-size: 1em;
  color: #333;
  background-color: transparent;

  padding: 10px 10px 10px 0;
  margin: 10px 10px 10px 0;

  appearance: none;

  &:focus {
    border-bottom-color: ${p => p.error ? 'palevioletred' : 'green'};
    outline: none;
  }
`
