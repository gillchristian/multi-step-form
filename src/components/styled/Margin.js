import styled from 'styled-components'

const marginCase = (side) => (p) => p[side] ? p[side] : ''

const Margin = styled.div`
  margin-right: ${marginCase('right')};
  margin-left: ${marginCase('left')};
  margin-top: ${marginCase('top')};
  margin-bottom: ${marginCase('bottom')};
`

export default Margin
