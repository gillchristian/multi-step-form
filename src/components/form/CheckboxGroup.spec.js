/* eslint-disable max-len */
import React from 'react'
import { shallow } from 'enzyme'
import toJSON from 'enzyme-to-json'

import CheckBoxGroup from './CheckboxGroup'
import { Checkbox } from '../styled'

const items = [
  { value: 'a1', label: 'Item A1' },
  { value: 'a2', label: 'Item A2' },
  { value: 'a3', label: 'Item A3' },
]


describe('<CheckBoxGroup /> component', () => {
  it('renders a list of Checkboxes', () => {
    const wrapper = shallow(
      <CheckBoxGroup
        items={items}
        value="a2"
        onChange={() => {}}
      />
    )

    const checkboxes = wrapper.find(Checkbox)

    expect(checkboxes.length).toBe(3)
    expect(toJSON(wrapper)).toMatchSnapshot()
  })

  describe('multiple values', () => {

    it('set the corresponding items as checked', () => {
      const onChange = jest.fn()
      const wrapper = shallow(
        <CheckBoxGroup
          items={items}
          value={['a2', 'a3']}
          onChange={onChange}
          multiple
        />
      )

      const checkboxes = wrapper.find(Checkbox)

      expect(checkboxes.at(0).prop('checked')).toBe(false)
      expect(checkboxes.at(1).prop('checked')).toBe(true)
      expect(checkboxes.at(2).prop('checked')).toBe(true)
    })

    it('calls onChange with the item value when a Checkbox is clicked', () => {
      const onChange = jest.fn()
      const wrapper = shallow(
        <CheckBoxGroup
          items={items}
          value={['a2']}
          onChange={onChange}
          multiple
        />
      )

      const checkboxes = wrapper.find(Checkbox)

      checkboxes.at(0).simulate('click')

      expect(onChange).toHaveBeenCalledTimes(1)
      expect(onChange).toHaveBeenCalledWith(['a2', 'a1'])
    })

    it('unchecks the clicked value if it was checked', () => {
      const onChange = jest.fn()
      const wrapper = shallow(
        <CheckBoxGroup
          items={items}
          value={['a2', 'a3']}
          onChange={onChange}
          multiple
        />
      )

      const checkboxes = wrapper.find(Checkbox)

      checkboxes.at(1).simulate('click')

      expect(onChange).toHaveBeenCalledTimes(1)
      expect(onChange).toHaveBeenCalledWith(['a3'])
    })

  })

  describe('single value', () => {

    it('set the corresponding item as checked', () => {
      const onChange = jest.fn()
      const wrapper = shallow(
        <CheckBoxGroup
          items={items}
          value="a2"
          onChange={onChange}
        />
      )

      const checkboxes = wrapper.find(Checkbox)

      expect(checkboxes.at(0).prop('checked')).toBe(false)
      expect(checkboxes.at(1).prop('checked')).toBe(true)
      expect(checkboxes.at(2).prop('checked')).toBe(false)
    })

    it('calls onChange with the item value when a Checkbox is clicked', () => {
      const onChange = jest.fn()
      const wrapper = shallow(
        <CheckBoxGroup
          items={items}
          value="a2"
          onChange={onChange}
        />
      )

      const checkboxes = wrapper.find(Checkbox)

      checkboxes.at(0).simulate('click')

      expect(onChange).toHaveBeenCalledTimes(1)
      expect(onChange).toHaveBeenCalledWith('a1')
    })

    it('calls onChange with the checked value if it is clicked', () => {
      const onChange = jest.fn()
      const wrapper = shallow(
        <CheckBoxGroup
          items={items}
          value="a2"
          onChange={onChange}
        />
      )

      const checkboxes = wrapper.find(Checkbox)

      checkboxes.at(1).simulate('click')

      expect(onChange).toHaveBeenCalledTimes(1)
      expect(onChange).toHaveBeenCalledWith('a2')
    })

  })

})
