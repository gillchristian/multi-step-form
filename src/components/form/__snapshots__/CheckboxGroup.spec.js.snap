exports[`<CheckBoxGroup /> component renders a list of Checkboxes 1`] = `
<div>
  <ListItem
    label="Item A1">
    <styled.div
      checked={false}
      onClick={[Function]} />
  </ListItem>
  <ListItem
    label="Item A2">
    <styled.div
      checked={true}
      onClick={[Function]} />
  </ListItem>
  <ListItem
    label="Item A3">
    <styled.div
      checked={false}
      onClick={[Function]} />
  </ListItem>
</div>
`;
